book/src/testnet-replicator.md

```diff
@@ -0,0 +1,200 @@
+and download the **Source Code** archive.  Extract the code and build the
+binaries with:
+```bash
+$ ./scripts/cargo-install-all.sh .
+$ export PATH=$PWD/bin:$PATH
+```
+
+### Starting The Replicator
+Try running following command to join the gossip network and view all the other nodes in the cluster:
+```bash
+$ solana-gossip --entrypoint testnet.solana.com:8001 spy
+# Press ^C to exit
+```
+
+##### Windows
+Now configure the keypairs for your replicator by running:
+
+Navigate to the solana install location and open a cmd prompt
+```bash
+$ solana-keygen.exe new -o replicator-keypair.json
+$ solana-keygen.exe new -o storage-keypair.json
+```
+
+Use solana-keygen to show the public keys for each of the keypairs, 
+they will be needed in the next step:
+```bash
+# The replicator's identity
+$ solana-keygen.exe pubkey replicator-keypair.json
+$ solana-keygen.exe pubkey storage-keypair.json
+```
+
+Then set up the storage accounts for your replicator by running:
+```bash
+$ solana-wallet.exe --keypair replicator-keypair.json airdrop 100000
+$ solana-wallet.exe --keypair replicator-keypair.json create-replicator-storage-account <replicator_pubkey> <storage_account_pubkey>
+```
+
+To start the replicator:
+```bash
+$ solana-replicator.exe --entrypoint testnet.solana.com:8001 --identity replicator-keypair.json --storage-keypair storage-keypair.json --ledger replicator-ledger
+```
+
+If the mining pool public key is known then it can be passed in to the replicator to enable
+automatic reward collection
+```bash
+$ solana-replicator.exe --entrypoint testnet.solana.com --identity replicator-keypair.json --storage-keypair storage-keypair.json --ledger replicator-ledger --mining-pool mining_pool_public_key
+```
+
+##### Linux and mac OS
+Setup the replicator and storage account keypairs
+```bash
+$ solana-keygen new -o ~/repliactor-keypair.json
+$ solana-keygen new -o ~/storage-keypair.json
+$ export REPLICATOR_IDENTITY=$(solana-keygen pubkey ~/replicator-keypair.json)
+$ export STORAGE_IDENTITY=$(solana-keygen pubkey ~/storage-keypair.json)
+
+```
+
+Then set up the storage accounts for your replicator by running:
+```bash
+$ solana-wallet --keypair ~/replicator-keypair.json airdrop 100
+$ solana-wallet --keypair ~/replicator-keypair.json create-replicator-storage-account $REPLICATOR_IDENTITY $STORAGE_IDENTITY
+
+```
+
+To start the replicator:
+```bash
+$ solana-replicator --entrypoint testnet.solana.com:8001 --identity ~/replicator-keypair.json --storage-keypair ~/storage-keypair.json --ledger replicator-ledger
+```
+
+If the mining pool public key is known then it can be passed in to the replicator to enable
+automatic reward collection
+```bash
+$ clear-config.sh
+$ solana-replicator --entrypoint testnet.solana.com --identity replicator-keypair.json --storage-keypair storage-keypair.json --ledger replicator-ledger --mining-pool mining_pool_public_key
+```
+
+If you built from source:
```

Comment:
I’d remove the “built from source” section for now, seems less important for somebody wanting to run a replicator